libs/island-ui/core/src/lib/TableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FC } from 'react'
+import { blue200 } from '@island.is/island-ui/theme'
+import { Text } from '../Text/Text'
+import { Box } from '../Box/Box'
+import { Stack } from '../Stack/Stack'
+
+interface Props {
+  title: string
+  headings: Array<string>
+  onClick: (heading: string) => void
+}
+
+export const TableOfContents: FC<Props> = ({ title, headings, onClick }) => (
+  <Box paddingX={4} paddingY={2} style={{ borderLeft: `1px solid ${blue200}` }}>
+    <Stack space={1}>
+      <Text variant="h5" as="h5">
+        {title}
+      </Text>
+      {headings.map((text, id) => (
+        <Box
+          key={id}
+          component="button"
+          type="button"
+          textAlign="left"
+          onClick={() => onClick(text)}
```

Comment:
will it be enough to only take in the text? won't we need some kind of anchor id or something?